megamek/src/megamek/client/bot/princess/BasicPathRanker.java

```diff
@@ -265,23 +262,27 @@ LosEffects calcLosEffects(IGame game, int shooterId, Targetable target) {
         }
 
         //  If they don't have LoS, they can't do damage.
-        LosEffects losEffects = calcLosEffects(game,
-                                               enemy.getId(),
-                                               path.getEntity());
+        LosEffects losEffects = 
+                LosEffects.calculateLos(game, enemy.getId(), path.getEntity(), shooterState.getPosition(), targetState.getPosition(), false);
+        
         if (!losEffects.canSee()) {
             return 0;
         }
-
-        if (targetState == null) {
-            targetState = new EntityState(path);
+        
+        Targetable actualTarget = path.getEntity();
+        
+        // if the target is infantry protected by a building, we have to fire at the building instead. 
+        if(losEffects.infantryProtected()) {
```

Comment:
This allows the bot to consider targeting infantry in a building for pathfinding purposes. She could shoot at them just fine, but would completely ignore them when evaluating paths, as they'd be considered "out of LOS".